pkg/manager/member/pd_failover.go

```diff
@@ -163,50 +164,45 @@ func (f *pdFailover) tryToDeleteAFailureMember(tc *v1alpha1.TidbCluster) error {
 		return err
 	}
 	// invoke deleteMember api to delete a member from the pd cluster
-	err = controller.GetPDClient(f.deps.PDControl, tc).DeleteMemberByID(memberID)
-	if err != nil {
-		klog.Errorf("pd failover: failed to delete member: %d, %v", memberID, err)
+	if err := controller.GetPDClient(f.deps.PDControl, tc).DeleteMemberByID(memberID); err != nil {
+		klog.Errorf("pd failover: failed to delete member %d, %v", memberID, err)
 		return err
 	}
-	klog.Infof("pd failover: delete member: %d successfully", memberID)
-	f.deps.Recorder.Eventf(tc, apiv1.EventTypeWarning, "PDMemberDeleted",
-		"%s(%d) deleted from cluster", failurePodName, memberID)
+	klog.Infof("pd failover: delete member %d successfully", memberID)
+	f.deps.Recorder.Eventf(tc, apiv1.EventTypeWarning, "PDMemberDeleted", "failure member %s(%d) deleted from PD cluster", failurePodName, memberID)
 
 	// The order of old PVC deleting and the new Pod creating is not guaranteed by Kubernetes.
 	// If new Pod is created before old PVC deleted, new Pod will reuse old PVC.
-	// So we must try to delete the PVC and Pod of this PD peer over and over,
-	// and let StatefulSet create the new PD peer with the same ordinal, but don't use the tombstone PV
+	// In this case, there is a period during which both new & old Pod is using the same PVC.
```

Comment:
This is impossible for most PVs because only a few kinds of PV support to be mounted by multiple Pods at the same time.
In addition, from the logic below, the old Pod will always be deleted first.